File: src/solar/solar-wc3/browser/Flag.ts
export default class Flag {

    //engine_config
    static ENGINE_DISABLE_GPU = 1 << 0;  // 关闭硬件加速  播放视频时不会黑屏闪烁
    static ENGINE_ENABLE_GPU_WEBGL = 1 << 1;  // 开启硬件加速  开启webgl 支持渲染模型 播放视频时，视频以外会黑屏闪烁 全屏播放即可
    static ENGINE_SINGLE_PROCESS = 1 << 2;  // 开启单进程模式    有时候多进程模式有bug的情况下 可以改换单进程模式  单进程会吃魔兽内存
    static ENGINE_ENABLE_COOKIES = 1 << 3;  // 开启cookie  开启后 会保存登陆账号 再次打开也会自动登陆


    //view_config
    static VIEW_NEW_WINDOW = 1 << 0;// 新窗口  会在魔兽上层覆盖新窗口，视频最好用新窗口 读图不卡
    static VIEW_DISABLE_TRANSPARENT = 1 << 1;// 关闭透明背景
}

File: scripts/solar/solar-wc3/browser/Flag.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 1,["7"] = 1,["8"] = 1,["10"] = 1,["11"] = 4,["12"] = 5,["13"] = 6,["14"] = 7,["15"] = 11,["16"] = 12});
local ____exports = {}
____exports.default = __TS__Class()
local Flag = ____exports.default
Flag.name = "Flag"
function Flag.prototype.____constructor(self)
end
Flag.ENGINE_DISABLE_GPU = 1 << 0
Flag.ENGINE_ENABLE_GPU_WEBGL = 1 << 1
Flag.ENGINE_SINGLE_PROCESS = 1 << 2
Flag.ENGINE_ENABLE_COOKIES = 1 << 3
Flag.VIEW_NEW_WINDOW = 1 << 0
Flag.VIEW_DISABLE_TRANSPARENT = 1 << 1
return ____exports
